### PR TITLE
ClangImporter: Fix quadratic behavior with property overrides 

### PR DESCRIPTION
**Pull request by slavapestov**
Fixes <rdar://problem/58363207>.

### CODE REVIEW

**Comment by slavapestov** [comment]
@swift-ci Please smoke test

**Comment by slavapestov** [comment]
@swift-ci Please smoke test

**Comment by slavapestov** [comment]
@swift-ci Please test source compatibility

**Comment by slavapestov** [comment]
@swift-ci Please smoke test

**Comment by slavapestov** [comment]
@swift-ci Please test source compatibility

**Comment by slavapestov** [comment]
@swift-ci Please smoke test Linux

File: lib/ClangImporter/ImportDecl.cpp [hunk 1]
```diff
@@ -2149,45 +2149,46 @@ namespace {
...
-          continue;
-
         if (auto *fd = dyn_cast<FuncDecl>(result)) {
           if (fd->isInstanceMember() != decl->isInstanceProperty())
             continue;
 
-          if (fd->getFullName().getArgumentNames().empty()) {
-            foundMethod = true;
-          }
+          assert(fd->getFullName().getArgumentNames().empty());
```
Comment (CodaFi) [inline lib/ClangImporter/ImportDecl.cpp]: This invariant feels sketchy to me.  The check above is not enough to guarantee this in general.

**Review by CodaFi — APPROVED** [review]
Barring one small comment, this looks great! Curious what the speedup is from this change alone?

File: lib/ClangImporter/ImportDecl.cpp (hunk 1 above)
Comment (slavapestov) [reply lib/ClangImporter/ImportDecl.cpp]: This invariant is enforced from when decls are added to MembersForNominal.

Comment (CodaFi) [reply lib/ClangImporter/ImportDecl.cpp]: Ah, I see.  Looks good then.

**Pull request merged**